programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -1597,13 +1597,52 @@ static int make_socket_path(uv_loop_t *loop) {
   uv_fs_t req;
   int rc, mkdir_rc;
 
-  rc = mkdir_rc = uv_fs_mkdir(loop, &req, SOCKET_PATH,
-        S_IRWXU|S_IRGRP|S_IXGRP, NULL);
+  mkdir_rc = uv_fs_mkdir(loop, &req, SOCKET_PATH, S_IRWXU|S_IRGRP|S_IXGRP, NULL);
   uv_fs_req_cleanup(&req);
 
-  if (rc != UV_ENOENT) {
+  if (mkdir_rc == 0 || mkdir_rc == UV_EEXIST) {
       rc = uv_fs_lstat(loop, &req, SOCKET_PATH, NULL);
       if (rc == 0) {
+          /* The directory will be owned by root:root (or root:wheel on bsd/darwin) if running as root/sudo. This leads
+           * to an unfortunate situation where users will need to be in the root (or wheel) group to use the sockets.
+           * We don't want to encourage users to run their UI (or whatever they are doing with the IPC sockets) with
+           * sudo; we also don't want to leave the IPC socket directory wide open.
+           *
+           * Look for a "ziti" user/group. If found:
+           *   - chgrp the directory to the ziti group
+           *   - log that the sockets can be used by users in the "ziti" group. include helpful `usermod` command
+           * If "ziti" group is not found:
+           *   - warn that IPC sockets are not being started because admin privs would be needed to access the sockets.
+           *   - include helpful commands to create ziti user/group
+           */
+          if (geteuid() == 0 && req.statbuf.st_gid == getegid()) {
```

Comment:
Should the ``chown`` be attempted when not root? What about the case that the user has ``ziti`` as a supplementary group?